olive/model/__init__.py

```diff
@@ -583,14 +577,28 @@ def get_dummy_inputs(self):
                 .to_data_container()
                 .get_first_batch()
             )
-        elif self.hf_config and self.hf_config.model_name:
-            dummy_inputs, _ = (
-                data_config_template.huggingface_data_config_template(
-                    self.hf_config.model_name,
-                    self.hf_config.dataset,
+        elif self.hf_config and self.hf_config.model_name and self.hf_config.task:
+            if self.hf_config.dataset:
+                logger.debug("Using hf_config.dataset to get dummy inputs")
+                dummy_inputs, _ = (
+                    data_config_template.huggingface_data_config_template(
+                        self.hf_config.model_name,
+                        self.hf_config.task,
+                        **self.hf_config.dataset,
+                    )
+                    .to_data_container()
+                    .get_first_batch()
                 )
-                .to_data_container()
-                .get_first_batch()
+            elif not self.hf_config.components:
+                logger.debug("Using hf onnx_config to get dummy inputs")
+                dummy_inputs = get_hf_model_dummy_input(
+                    self.hf_config.model_name, self.hf_config.task, self.hf_config.feature
+                )
+
+        if dummy_inputs is None:
+            raise ValueError(
+                "Unable to get dummy inputs. Please provide dummy_inputs_func, io_config.input_shapes,"
+                " hf_config.dataset, or hf_config."
             )
```

Comment:
need we add unit test for these logic?

Reply:
Just pushed a commit with new unit tests for `PyTorchModel.get_dummy_inputs`